### ISSUE TITLE
Only backup when certain conditions are met

### ISSUE THREAD

**Issue by Bixilon** [signalapp/Signal-Android#13007]
So, feedback about the (changed!) backup scheudler:
Backups are now always done, no matter in what state the device is.

- I really don't need backups every day, it just creates a big file every day, I think weekly backup is enough
- Backups are also done when ac is disconnected (that actually drains my battery a lot)
- Backups are done at random time, somtimes they start exactly when I want to use my phone, thus slowing it down. Nightly is the best option here!

Please change those 3 points and a lot of users will thank you!

Thanks

Moritz

**Comment by Giga-Pudding** [comment]
Another point:

I fear wearout of the eMMC flashmemory in my smartphone! I only use Signal for a few contacts and one groupchat, so overall not that much. However, my backup file is already ~4GB. A Backup everyday results in 1,46 terabytes written per year. Imagine a person that uses Signal more than me... let's assume a backup size of 15GB: that makes 5,4 Terabyte written each year!

I honestly don't know how much terabytes can be written on todays eMMC memory, but i fear worn out storage and a loss of data. How about backups without media files included? That would reduce the filesize alot!

Perfect would be:
- daily backup without media files
- backup including media files once per week or once every two weeks

BR!

**Comment by Bixilon** [comment]
@Giga-Pudding That is also a great concern, just recently an sd card died and a lot of important data was lost.

**Comment by Bixilon** [comment]
> We used to be able to require AC, but it sort of doesn't jive super-well with our timed backups. I think for most people, if they pick a time overnight, their phone will be charging, and it's sort of a moot point.


I know, I still prefer that over the current situation. Please add it as an option like "Only on ac"


> We let you choose your backup time in the backup settings, but there is a caveat here: sometimes Android just doesn't want to respect our alarms to wake up the app. We've definitely seen some devices where an alarm is set for 2am and Android only tells us about it at 8am when the user starts using their phone again, which isn't what we want! Unfortunately, we're using the "most exact" type of alarm that makes sense for this scenario. The only way we could have a stronger guarantee is to use the Alarm Clock APIs, which we do use for scheduled messages, but have the unfortunate consequence of appearing as "upcoming alarms" in other areas of the android system UI, and therefore don't make sense for the backup case. Not a ton we can do here :( Maybe disabling battery optimizations will help?

Disabling it will certainly help and that is exactly what is causing it. The problem is, that most people don't want to disable them (for a good reason) and they normally plug off their phone, before using it. And I am charging my phone just every 3 days, so 2-3 backups run without ac being conencted and I just want to start using it.



> [greyson-signal](https://github.com/greyson-signal) closed this as [completed](https://github.com/signalapp/Signal-Android/issues?q=is%3Aissue+is%3Aclosed+archived%3Afalse+reason%3Acompleted) [6 minutes ago](https://github.com/signalapp/Signal-Android/issues/13007#event-9593425412)


I don't see why you closed this. This is still an ongoing problem that needs to be solved.

**Comment by Bixilon** [comment]
So, is this now on a needs to be done list or something? Do I need to do something else?

**Comment by Giga-Pudding** [comment]
To be honest:
I do consider unnecessary wearout of the eMMC flashmemory as bug. If you save your backup on a MicroSD card instead of the internal eMMC storage, the problem gets even worse, since the endurance of most MicroSD cards is much smaller. RaspberryPI owners know this problem well, and i'm sure most Raspberries do not even write multiple gigabytes per day.

There needs to be a way to make the backup smaller (for example by excluding media files like pictures or videos from the backup).

Please allow us to make "chat-only"-backups every day, and "full"-backups including media files every other week.

**Comment by greyson-signal** [comment]
> So, is this now on a needs to be done list or something? Do I need to do something else?

Yeah, we're aware of it. You don't need to do anything else. Signal does not have a public roadmap or anything. We reserve the right to work on things in the order we think is best.

> Please allow us to make "chat-only"-backups every day, and "full"-backups including media files every other week.

I appreciate the suggestion, but just to reiterate, we use github for bug tracking, not feature suggestions. But this is something we'll think about.
